File: PTA/Advanced_Level/1061.cpp
#include<bits/stdc++.h>
using namespace std;
string week[] = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"};
int main() {
    string a, b, c, d;
        cin >> a >> b >> c >> d;
    int k = 0;
    while(true) {
        if(a[k] == b[k] && a[k] >= 'A' && a[k] <= 'G') break;
        ++k; 
    } 
    cout << week[a[k] - 'A'] << ' ';
    k++;
    while(true) {
        if(a[k] == b[k] && (a[k] >= '0' && a[k] <= '9' || a[k] >= 'A' && a[k] <= 'N')) break;
        ++k;
    }
    printf("%02d:", a[k] <= '9' ? a[k] - '0' : a[k] - 'A' + 10);
    for(int i = 0; ; i++) {
        if(c[i] == d[i] && isalpha(c[i])) {
            printf("%02d", i);
            break;
        }

    }
    return 0;
}

File: PTA/Advanced_Level/1061.rs
use std::io::stdin;

fn main() {
    let week: Vec<&str> = vec!["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
    let mut input: String = String::new();
    stdin().read_line(&mut input).unwrap();
    let t: Vec<String> = input.trim().split(' ').map(|s| s.to_string()).collect();
    let mut v: vec![Vec<char>; 4] = vec![];
    let mut index: usize = 0;
    for x in t {
        let v1: Vec<char> = x.chars().collect();
        v.push(v1.clone());
    }
    println!("{:?}", v);
    /*while true {
        if v[0][index] == v[1][index] && v[0][index] >= 'A' && v[0][index] <= 'G' {
            break;
        } 
        index += 1;
    }
    */
    //println!("{} ", week[(v[0][index] as usize - 'A' as usize)]);
}